include/envoy/server/filter_config.h

```diff
@@ -151,6 +154,24 @@ class NamedNetworkFilterConfigFactory {
                                                      FactoryContext& context) PURE;
 
   /**
+   * v2 variant of createFilterFactory(..), where filter configs are specified as proto. This may be
+   * optionally implemented today, but will in the future become compulsory once v1 is deprecated.
+   */
+  virtual NetworkFilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message& config,
+                                                              FactoryContext& context) {
+    UNREFERENCED_PARAMETER(config);
+    UNREFERENCED_PARAMETER(context);
+    return NetworkFilterFactoryCb();
+  }
+
+  /**
+   * @return std::unique_ptr<Protobuf::Message> create empty config proto message for v2. The filter
+   *         config will be parsed into the result of this method. Optional today, will be
+   *         compulsory when v1 is deprecated.
+   */
+  virtual std::unique_ptr<Protobuf::Message> createConfigProto() { return nullptr; }
```

Comment:
What you say above is correct, except for the bit about the "type inside". There's no implied type in a `google.protobuf.Struct`. If we had gone down the route of `google.protobuf.Any`, there would be a type URL present in the JSON that we could use to perform this conversion without the filter having to supply this factory (see the other comment on this topic for more  discussion of JSON->message).